bpf/bpf_lxc.c

```diff
@@ -258,8 +258,7 @@ static inline int ipv6_l3_from_lxc(struct __sk_buff *skb,
 		/* After L4 write in port mapping: revalidate for direct packet access */
 		data = (void *) (long) skb->data;
 		data_end = (void *) (long) skb->data_end;
-		ip6 = data + ETH_HLEN;
-		if (data + sizeof(*ip6) + ETH_HLEN > data_end)
+		if (data + sizeof(*(data + ETH_HLEN)) + ETH_HLEN > data_end)
```

Comment:
I think we can remove this length check altogether, it looks like code that was accessing ip6 before was removed without removing this check.